tools/coqdoc/output.ml

```diff
@@ -260,6 +260,40 @@ module Latex = struct
   let label_ident s =
     for i = 0 to String.length s - 1 do label_char s.[i] done
 
+
+  let convert_string f s =
+    let n = String.length s in
+    let rec qmap i =
+      if i == n then "" else (f s.[i]) ^ (qmap (i + 1))
+    in qmap 0
+
+  let fragment_char c =
+      match c with
+        | 'a'..'z' | 'A'..'Z' | '0'..'9' | '.' | '_'
+             | '<' | '>' | '&' | '\'' | '"' | '-' | ':' -> true
+        | _ -> false
+
+   let not_fragment_char c = not (fragment_char c)
+
+   let string_contains f s =
+     let n = String.length s in
+     let rec find i =
+        if i == n then false else if f s.[i] then true else find (i+1)
+      in find 0
+
+  let encode_fragment s =
+     if string_contains (not_fragment_char) s
+     then Digest.to_hex (Digest.string s)
```

Comment:
What I said in #13787 still holds. The fragment of the URL in the LaTeX backend has to be identical to the one in the HTML backend. (They point to the exact same object on the Internet.) So, please use the same function for both backends. Do not let the code give the impression they are unrelated.